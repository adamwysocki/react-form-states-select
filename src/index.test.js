// index.test.js
import React from "react";
import App from "./index";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import sinon, { stub } from "sinon";
import Adapter from "enzyme-adapter-react-16";

// React 16 Enzyme Adapter
Enzyme.configure({ adapter: new Adapter() });

it("should render", () => {
  const onStateSelect = stub();
  const component = renderer.create(<App onChange={onStateSelect} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render without a default option if default option is turned off", () => {
  const onStateSelect = stub();
  const wrapper = mount(<App onChange={onStateSelect} hasDefaultOption={false} />);
  const select = wrapper.find("select");
  expect(select.contains(<option value="null">Select a state ...</option>)).toEqual(false);
});

it("should render with the default option text specified", () => {
  const onStateSelect = stub();
  const wrapper = mount(<App onChange={onStateSelect} defaultOptionText={"State ..."} />);
  const select = wrapper.find("select");
  expect(select.contains(<option value="null">State ...</option>)).toEqual(true);
});

it("should render with a custom formatted label", () => {
  const onStateSelect = stub();
  const component = renderer.create(<App onChange={onStateSelect} valueTemplate={"[ABBR] - [STATE]"} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
