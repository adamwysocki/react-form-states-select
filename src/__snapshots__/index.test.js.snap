// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="sc-bdVaJa fXNkFd"
>
  <select
    className=""
    id="state"
    name="state"
    onChange={[Function]}
    style={Object {}}
  >
    <option
      value="null"
    >
      Select a state ...
    </option>
    <option
      value={0}
    >
      Alabama
    </option>
    <option
      value={1}
    >
      Alaska
    </option>
    <option
      value={2}
    >
      American Samoa
    </option>
    <option
      value={3}
    >
      Arizona
    </option>
    <option
      value={4}
    >
      Arkansas
    </option>
    <option
      value={5}
    >
      California
    </option>
    <option
      value={6}
    >
      Colorado
    </option>
    <option
      value={7}
    >
      Connecticut
    </option>
    <option
      value={8}
    >
      Delaware
    </option>
    <option
      value={9}
    >
      District Of Columbia
    </option>
    <option
      value={10}
    >
      Federated States Of Micronesia
    </option>
    <option
      value={11}
    >
      Florida
    </option>
    <option
      value={12}
    >
      Georgia
    </option>
    <option
      value={13}
    >
      Guam
    </option>
    <option
      value={14}
    >
      Hawaii
    </option>
    <option
      value={15}
    >
      Idaho
    </option>
    <option
      value={16}
    >
      Illinois
    </option>
    <option
      value={17}
    >
      Indiana
    </option>
    <option
      value={18}
    >
      Iowa
    </option>
    <option
      value={19}
    >
      Kansas
    </option>
    <option
      value={20}
    >
      Kentucky
    </option>
    <option
      value={21}
    >
      Louisiana
    </option>
    <option
      value={22}
    >
      Maine
    </option>
    <option
      value={23}
    >
      Marshall Islands
    </option>
    <option
      value={24}
    >
      Maryland
    </option>
    <option
      value={25}
    >
      Massachusetts
    </option>
    <option
      value={26}
    >
      Michigan
    </option>
    <option
      value={27}
    >
      Minnesota
    </option>
    <option
      value={28}
    >
      Mississippi
    </option>
    <option
      value={29}
    >
      Missouri
    </option>
    <option
      value={30}
    >
      Montana
    </option>
    <option
      value={31}
    >
      Nebraska
    </option>
    <option
      value={32}
    >
      Nevada
    </option>
    <option
      value={33}
    >
      New Hampshire
    </option>
    <option
      value={34}
    >
      New Jersey
    </option>
    <option
      value={35}
    >
      New Mexico
    </option>
    <option
      value={36}
    >
      New York
    </option>
    <option
      value={37}
    >
      North Carolina
    </option>
    <option
      value={38}
    >
      North Dakota
    </option>
    <option
      value={39}
    >
      Northern Mariana Islands
    </option>
    <option
      value={40}
    >
      Ohio
    </option>
    <option
      value={41}
    >
      Oklahoma
    </option>
    <option
      value={42}
    >
      Oregon
    </option>
    <option
      value={43}
    >
      Palau
    </option>
    <option
      value={44}
    >
      Pennsylvania
    </option>
    <option
      value={45}
    >
      Puerto Rico
    </option>
    <option
      value={46}
    >
      Rhode Island
    </option>
    <option
      value={47}
    >
      South Carolina
    </option>
    <option
      value={48}
    >
      South Dakota
    </option>
    <option
      value={49}
    >
      Tennessee
    </option>
    <option
      value={50}
    >
      Texas
    </option>
    <option
      value={51}
    >
      Utah
    </option>
    <option
      value={52}
    >
      Vermont
    </option>
    <option
      value={53}
    >
      Virgin Islands
    </option>
    <option
      value={54}
    >
      Virginia
    </option>
    <option
      value={55}
    >
      Washington
    </option>
    <option
      value={56}
    >
      West Virginia
    </option>
    <option
      value={57}
    >
      Wisconsin
    </option>
    <option
      value={58}
    >
      Wyoming
    </option>
  </select>
</div>
`;

exports[`should render with a custom formatted label 1`] = `
<div
  className="sc-bxivhb kAlGYy"
>
  <select
    className=""
    id="state"
    name="state"
    onChange={[Function]}
    style={Object {}}
  >
    <option
      value="null"
    >
      Select a state ...
    </option>
    <option
      value={0}
    >
      AL - [STATE]
    </option>
    <option
      value={1}
    >
      AK - [STATE]
    </option>
    <option
      value={2}
    >
      AS - [STATE]
    </option>
    <option
      value={3}
    >
      AZ - [STATE]
    </option>
    <option
      value={4}
    >
      AR - [STATE]
    </option>
    <option
      value={5}
    >
      CA - [STATE]
    </option>
    <option
      value={6}
    >
      CO - [STATE]
    </option>
    <option
      value={7}
    >
      CT - [STATE]
    </option>
    <option
      value={8}
    >
      DE - [STATE]
    </option>
    <option
      value={9}
    >
      DC - [STATE]
    </option>
    <option
      value={10}
    >
      FM - [STATE]
    </option>
    <option
      value={11}
    >
      FL - [STATE]
    </option>
    <option
      value={12}
    >
      GA - [STATE]
    </option>
    <option
      value={13}
    >
      GU - [STATE]
    </option>
    <option
      value={14}
    >
      HI - [STATE]
    </option>
    <option
      value={15}
    >
      ID - [STATE]
    </option>
    <option
      value={16}
    >
      IL - [STATE]
    </option>
    <option
      value={17}
    >
      IN - [STATE]
    </option>
    <option
      value={18}
    >
      IA - [STATE]
    </option>
    <option
      value={19}
    >
      KS - [STATE]
    </option>
    <option
      value={20}
    >
      KY - [STATE]
    </option>
    <option
      value={21}
    >
      LA - [STATE]
    </option>
    <option
      value={22}
    >
      ME - [STATE]
    </option>
    <option
      value={23}
    >
      MH - [STATE]
    </option>
    <option
      value={24}
    >
      MD - [STATE]
    </option>
    <option
      value={25}
    >
      MA - [STATE]
    </option>
    <option
      value={26}
    >
      MI - [STATE]
    </option>
    <option
      value={27}
    >
      MN - [STATE]
    </option>
    <option
      value={28}
    >
      MS - [STATE]
    </option>
    <option
      value={29}
    >
      MO - [STATE]
    </option>
    <option
      value={30}
    >
      MT - [STATE]
    </option>
    <option
      value={31}
    >
      NE - [STATE]
    </option>
    <option
      value={32}
    >
      NV - [STATE]
    </option>
    <option
      value={33}
    >
      NH - [STATE]
    </option>
    <option
      value={34}
    >
      NJ - [STATE]
    </option>
    <option
      value={35}
    >
      NM - [STATE]
    </option>
    <option
      value={36}
    >
      NY - [STATE]
    </option>
    <option
      value={37}
    >
      NC - [STATE]
    </option>
    <option
      value={38}
    >
      ND - [STATE]
    </option>
    <option
      value={39}
    >
      MP - [STATE]
    </option>
    <option
      value={40}
    >
      OH - [STATE]
    </option>
    <option
      value={41}
    >
      OK - [STATE]
    </option>
    <option
      value={42}
    >
      OR - [STATE]
    </option>
    <option
      value={43}
    >
      PW - [STATE]
    </option>
    <option
      value={44}
    >
      PA - [STATE]
    </option>
    <option
      value={45}
    >
      PR - [STATE]
    </option>
    <option
      value={46}
    >
      RI - [STATE]
    </option>
    <option
      value={47}
    >
      SC - [STATE]
    </option>
    <option
      value={48}
    >
      SD - [STATE]
    </option>
    <option
      value={49}
    >
      TN - [STATE]
    </option>
    <option
      value={50}
    >
      TX - [STATE]
    </option>
    <option
      value={51}
    >
      UT - [STATE]
    </option>
    <option
      value={52}
    >
      VT - [STATE]
    </option>
    <option
      value={53}
    >
      VI - [STATE]
    </option>
    <option
      value={54}
    >
      VA - [STATE]
    </option>
    <option
      value={55}
    >
      WA - [STATE]
    </option>
    <option
      value={56}
    >
      WV - [STATE]
    </option>
    <option
      value={57}
    >
      WI - [STATE]
    </option>
    <option
      value={58}
    >
      WY - [STATE]
    </option>
  </select>
</div>
`;
